modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/direct/TransportPutDatabaseConfigurationAction.java

```diff
@@ -160,9 +159,10 @@ ClusterState execute(ClusterState currentState) throws Exception {
                 logger.debug("updating existing database configuration [{}]", id);
             }
 
-            Metadata currentMeta = currentState.metadata();
             return ClusterState.builder(currentState)
-                .metadata(Metadata.builder(currentMeta).putCustom(IngestGeoIpMetadata.TYPE, geoIpMeta))
+                .putProjectMetadata(
+                    ProjectMetadata.builder(project).putCustom(IngestGeoIpMetadata.TYPE, new IngestGeoIpMetadata(databases))
```

Comment:
I think we can still use `geoIpMeta` here since it is updated on line 154?

Reply:
Great catch! Thanks